tools/log-collector/helper.go

```diff
@@ -335,3 +270,21 @@ func checkLabelExist(givenLabel, toCheckInLabel map[string]string) (exist bool)
 	}
 	return exist
 }
+
+// filterObjectsOnLabel filter objects on the basis of Labels
+func filterObjectsOnLabel(allObjects unstructured.UnstructuredList) (objects unstructured.UnstructuredList) {
+
+	for _, object := range allObjects.Items {
+		objectLabel := object.GetLabels()
+		if len(objectLabel) != 0 && checkLabelExist(objectLabel, K8STrilioVaultLabel) {
+			objects.Items = append(objects.Items, object)
+		}
+		if nonLabeledResources.Has(object.GetKind()) {
+			objects.Items = append(objects.Items, object)
+		}
+		if clusteredResources.Has(object.GetKind()) {
+			objects.Items = append(objects.Items, object)
+		}
```

Comment:
These two checks can be moved to this func's parent func as allObjects belong to same kind right?
add something like `if nonLabeledResources.Has(resource.Name)` don't filter resource list

Reply:
we are doing the resources filtering on parent's function and here and passing all object of that filtered resources and doing the label check in the end. So it will take only Labeled Resources from that filter,

Reply:
resolved